src/unzip.rs

```diff
@@ -0,0 +1,68 @@
+use crate::{error::Result, jobs, http::HTTP_CLIENT};
+use reqwest::blocking::RequestBuilder;
+use std::fs;
+use std::io::Write;
+use std::path::Path;
+use zip::ZipArchive;
+
+struct UnzipPrep {
+    req: RequestBuilder,
+    unzip_directory: String,
+}
+
+fn construct_unzip(
+    url: &str,
+    unzip_directory: &str
+) -> UnzipPrep {
+    
+    let req = HTTP_CLIENT.get(url);
+    let dir_copy = unzip_directory.to_string();
+    
+    UnzipPrep {
+        req,
+        unzip_directory: dir_copy,
+    }
+}
+
+byond_fn! { unzip_download_async(url, unzip_directory) {
+    let unzip = construct_unzip(&url, &unzip_directory);
+    Some(jobs::start(move ||
+        do_unzip_download(unzip).unwrap_or_else(|e| e.to_string())
+    ))
+} }
+
+fn do_unzip_download(prep: UnzipPrep) -> Result<String> {
+    let unzip_path = Path::new(&prep.unzip_directory);
+    let response = prep.req.send()?;
+
+    let content = response.bytes()?;
+
+    let reader = std::io::Cursor::new(content);
+    let mut archive = ZipArchive::new(reader)?;
+
+    for i in 0..archive.len()
+    {
+        let mut entry = archive.by_index(i).unwrap();
```

Comment:
```suggestion
        let mut entry = archive.by_index(i)?;
```
This and handle the error instead of unwrapping and potentially panicking?